addons/account/models/product.py

```diff
@@ -118,6 +118,22 @@ def _check_uom_not_in_invoice(self):
                 "If you want to change its Unit of Measure, please archive this product and create a new one."
             ))
 
+    @api.model
+    def _force_default_tax(self, products, companies):
+        if not products:
+            # in case we're calling that function from demo data file, we want to force the default taxes on all the templates
+            products = self.env['product.template'].search([])
```

Comment:
Why not already filter the products without companies here? Might help regarding perfs in some cases

Reply:
I supposed that theoretically (there is no usage for that yet, nor perf issues), we want that function to address all the products the user can see, that's why I left the record rules do that filtering. 

For example in a multi branch environment, the product might have a parent company set, yet we'd might want the default tax of the branch to be forced there as well.

Reply:
It doesn't really make sense to me.
But the way, why don't we use this function in the loading of the CoA also? Looks like duplicated code.

Reply:
well it's not exactly the same code since there we want the default tax only on products that have at least a tax in another company, because some modules/features are working like that (they set a product in data that doesn't intent to ever have any tax, eg. TIPS in PoS). 

But when you create a product, you don't have such a reference point :shrug:

Reply:
Just call filtered (on `taxes_id`)  inside the create, it will be more explicit on top of it that you are forcing default taxes only on products created without taxes (set to False explicitly)

Reply:
I don't get your point :thinking:  but the recordset operation was buggy, so I had to refactor anyway, and ended up in a code in chart_template.py that calls product.py. So you may find it ok like that?

Reply:
It would if run it wasn't red 😅🙈

Reply:
it's not anymore... :smirk: